packages/libs/deer/tests/test_impls_core_non_zero.rs

```diff
@@ -0,0 +1,108 @@
+// These values use the same logic as integral, but zero is an error!
+
+mod common;
+
+use core::num::{
+    NonZeroI128, NonZeroI16, NonZeroI32, NonZeroI64, NonZeroI8, NonZeroIsize, NonZeroU128,
+    NonZeroU16, NonZeroU32, NonZeroU64, NonZeroU8, NonZeroUsize,
+};
+
+use deer::{Deserialize, Number};
+use deer_desert::{assert_tokens_error, Token};
+use serde_json::json;
+
+macro_rules! test_zero {
+    ($ty:ident) => {
+        paste::paste! {
+            #[test]
+            fn [<$ty:lower _err_zero >]() {
+                let zero = Number::from(0u8);
+
+                assert_tokens_error::<_, $ty>(
+                    &error! {
+                        ns: "deer",
+                        id: ["value"],
+                        properties: {
+                            "expected": $ty::reflection(),
+                            "received": 0,
+                            "location": []
+                        }
+                    },
+                    &[Token::Number(zero)],
+                )
+            }
+        }
+    };
+
+    [$($ty:ident),* $(,)?] => {
+        $(test_zero!($ty);)*
+    };
+}
+
+test_zero![
+    NonZeroU8, NonZeroU16, NonZeroU32, NonZeroU64, NonZeroI8, NonZeroI16, NonZeroI32, NonZeroI64,
+];
+
+#[test]
+fn i128_err_zero() {
```

Comment:
I assume u128/i128/isize/usize can't be tested with the macro above?

Reply:
Yes, `Number` (without the `arbitrary-precision` feature) cannot represent values with a greater width than 8 bytes. This is a trade-off, as one can expect that `u64` is a lot more common than `u128` and the space occupied by `u128` is twice as large. This is also why there are four additional tokens for these types.